components/script/dom/windowproxy.rs

```diff
@@ -386,7 +386,14 @@ impl WindowProxy {
                     opener_id,
                 ) {
                     Some(opener_top_id) => {
-                        let global_to_clone_from = unsafe { GlobalScope::from_context(cx) };
+                        let global_to_clone_from = unsafe {
+                            let in_compartment_proof =
+                                AlreadyInCompartment::assert_for_cx(SafeJSContext::from_ptr(cx));
```

Comment:
Instead, let's add an `InCompartment` argument to `opener`, and since it's only called by Window::Opener we can modify https://github.com/servo/servo/blob/12693b51f510761d5dd58a9103596a192edb803c/components/script/dom/bindings/codegen/Bindings.conf#L60 to include that method.